File: app/src/main/java/activity/TvShowsActivity.java
package activity;

/**
 * Created by icaro on 14/09/16.
 */

import android.app.SearchManager;
import android.content.Context;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.support.v7.widget.SearchView;
import android.view.Menu;
import android.view.MenuItem;

import br.com.icaro.filme.R;
import fragment.TvShowsFragment;
import utils.Constantes;


public class TvShowsActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list_main); // ???
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        setUpToolBar();
        setupNavDrawer();
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setTitle(getIntent()
                .getStringExtra(Constantes.INSTANCE.getNAV_DRAW_ESCOLIDO()));

//        AdView adview = (AdView) findViewById(R.id.adView);
//        AdRequest adRequest = new AdRequest.Builder()
//                .addTestDevice(AdRequest.DEVICE_ID_EMULATOR)        // All emulators
//                .addTestDevice("AC98C820A50B4AD8A2106EDE96FB87D4")  // An example device ID
//                .build();
//        adview.loadAd(adRequest);

        if (savedInstanceState == null) {
            TvShowsFragment tvShowsFragment = new TvShowsFragment();
            tvShowsFragment.setArguments(getIntent().getExtras());
            setCheckable(getIntent().getIntExtra(Constantes.INSTANCE.getABA(), 0));
            getSupportFragmentManager()
                    .beginTransaction()
                    .add(R.id.container_list_main, tvShowsFragment)
                    .commit();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu, menu);

        SearchView searchView = (SearchView) menu.findItem(R.id.search).getActionView();
        SearchManager searchManager = (SearchManager) getSystemService(Context.SEARCH_SERVICE);
        searchView.setSearchableInfo(searchManager.getSearchableInfo(getComponentName()));
        searchView.setQueryHint(getResources().getString(R.string.procurar));
        searchView.setEnabled(false);

        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId()) {

            case android.R.id.home: {
                finish();
                break;
            }
        }

        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/listaserie/activity/TvShowsActivity.kt
package listaserie.activity

/**
 * Created by icaro on 14/09/16.
 */

import activity.BaseActivity
import android.app.SearchManager
import android.content.Context
import android.content.pm.ActivityInfo
import android.os.Bundle
import android.support.v7.widget.SearchView
import android.view.Menu
import android.view.MenuItem
import br.com.icaro.filme.R
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.AdView
import listaserie.fragment.TvShowsFragment
import utils.Constantes


class TvShowsActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_list_main) // ???
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        setUpToolBar()
        setupNavDrawer()
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        val titulo = resources.getString(intent
                .getIntExtra(Constantes.NAV_DRAW_ESCOLIDO, 0))
        supportActionBar?.title = titulo

        val adview = findViewById<AdView>(R.id.adView)
        val adRequest = AdRequest.Builder()
                .addTestDevice(AdRequest.DEVICE_ID_EMULATOR)        // All emulators
                .addTestDevice("AC98C820A50B4AD8A2106EDE96FB87D4")  // An example device ID
                .build()
        adview.loadAd(adRequest)

        if (savedInstanceState == null) {
            val tvShowsFragment = TvShowsFragment()
            tvShowsFragment.arguments = intent.extras
            setCheckable(intent.getIntExtra(Constantes.ABA, 0))
            supportFragmentManager
                    .beginTransaction()
                    .add(R.id.container_list_main, tvShowsFragment)
                    .commit()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        val searchView = menu.findItem(R.id.search).actionView as SearchView
        val searchManager = getSystemService(Context.SEARCH_SERVICE) as SearchManager
        searchView.setSearchableInfo(searchManager.getSearchableInfo(componentName))
        searchView.queryHint = resources.getString(R.string.procurar)
        searchView.isEnabled = false
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        when (item.itemId) {
            android.R.id.home -> {
                finish()
            }
        }
        return super.onOptionsItemSelected(item)
    }
}

